### PR TITLE
Feature/ibms

### PR DESCRIPTION
**Pull request by BillyAbildgaard**
merging feature branch with master.  Doing this to get all of the commits from master into the feature branch so we will be ready to merge in the master branch at the end of this sprint

Needed to preserve changes made to lib/models/obms

moved changes in lib/models/obms to lib/models/network-management.  This is the only file I manually touched if you want to review this, specifically.

### CODE REVIEW

File: lib/models/network-management.js [hunk 1]
```diff
@@ -160,7 +163,12 @@ function NetworkManagementModelFactory (
                 assert.ok(mgmtSettings.length === 0 || mgmtSettings.length === 1,
                     "mgmts should be unique for a given node and service."
                 );
-                return _.first(mgmtSettings);
+                {
+                    if (mgmtSetting) {
```
Comment (brianparry) [inline lib/models/network-management.js]: I think this should be mgmtSettings (plural)

**Comment by tldavies** [comment]
👎  

There should not be 49 commits with most of them already in master. Please rebase on master before merging.

**Comment by brianparry** [comment]
Hi @tldavies.  I agree that this PR is very difficult to review, but as we discussed, merging 49 commits from master into `feature/ibms` is exactly what we want to do.  Rebasing rewrites history which requires a force push and forces everyone downstream to interactively rebase their work. @BillyAbildgaard can point you to the relevant changes from master (resolution of merge conflicts).

**Comment by BillyAbildgaard** [comment]
@tldavies after our discussion with @brianparry offline last week do you still think this PR should have a -1?

**Comment by panpan0000** [comment]
Please rebase your PR branch to latest master , otherwise, on-tasks unit-test will fail. (task-schema)

**Comment by BillyAbildgaard** [comment]
@panpan0000 just pulled in the latest changes from master.  As a reminder, all of these commits will not show up when we go to merge the feature branch into master.  At that point, only our new commits will be seen

**Pull request merged**